src/monattack.cpp

```diff
@@ -932,6 +968,9 @@ void mattack::fungus_haze(monster *z, int index)
 
 void mattack::fungus_big_blossom(monster *z, int index)
 {
+    if( z->friendly ) {
+        return; // TODO: handle friendly monsters
```

Comment:
A friendly giant blossom would likewise not change its behavior for Mycus folks, and I've no idea why a Mycus would otherwise be friendly.  (Absent weird artifact trancing, and then it thinks you're a Mycus.)